timescaledb/load_data.py

```diff
@@ -0,0 +1,245 @@
+            id SERIAL,
+            time TIMESTAMPTZ,
+            db_version TEXT,
+            branch TEXT,
+            query_mode TEXT,
+            build_id TEXT,
+            git_commit_id TEXT,
+            benchmark_type TEXT,
+            scalefactor NUMERIC,
+            terminals NUMERIC,
+            weight JSONB, 
+            result JSONB
+        );
+    """ % TABLENAME
+    INSTALL_TIMESCALE = """
+        CREATE EXTENSION IF NOT EXISTS timescaledb CASCADE;
+    """
+    CONVERT_TO_HYPERTABLE_SQL = """
+        SELECT create_hypertable('%s', 'time');
+    """ % TABLENAME
+    CREATE_INDEX_SQL = """
+        CREATE INDEX ON %s (git_commit_id, time DESC);
+    """ % TABLENAME
+    cur = conn.cursor()
+    cur.execute(CREATE_SQL)
+    cur.execute(INSTALL_TIMESCALE)
+    cur.execute(CONVERT_TO_HYPERTABLE_SQL)
+    cur.execute(CREATE_INDEX_SQL)
+    conn.commit()
+    cur.close()
+
+
+def read_and_insert_from_folder(conn, path):
+    """Read data from folder and insert those data into database.
+
+    Currently this only support the structure as show in test_data. If we want
+    to read data from folders with other structure, we need to re-write the
+    reading part.
+
+    Args:
+        conn (psycopg2.extensions.connection): The connection to PostgreSQL database.
+        path (str): The parent destination folder.
+
+    """
+    # TODO(benliangw): These data are not included in expconfig and summary file.
+    branch = 'master'
+    query_mode = 'extended'
+    build_id = 35
+    git_commit_id = '4ed4617'
+
+    for subdir, dirs, files in os.walk(path):
+        if subdir != str(path):
+            benchmark_type, timestamp, db_version, throughput, scalefactor, terminals = read_from_summary(
+                os.path.join(subdir, 'oltpbench.summary'))
+            weight = read_from_expconfig(
+                os.path.join(subdir, 'oltpbench.expconfig'))
+            insert_data(conn, time=timestamp, db_version=db_version, 
+                        branch=branch, query_mode=query_mode, build_id=35, 
+                        git_commit_id=git_commit_id, benchmark_type=benchmark_type, 
+                        scalefactor=scalefactor, terminals=terminals, weight=weight, 
+                        result=json.dumps({'throughput': throughput}))
+
+
+def read_from_summary(path):
+    """Read data from file ends with ".summary".
+
+    Args:
+        path (str): The position of the summary file.
+        
+    Returns:
+        benchmark_type (str): The type of benchmark. It could be noop/smallbank/tatp
+        timestamp (str): The timestamp when the benchmark was created in milliseconds.
+        db_version (str): The version of NoisePage Database.
+        throughput (str): The throughtput calculated by benchmark testing.
+        scalefactor (str): The scalefactor used for benchmark testing.
+        terminals (str): The number of terminals used for benchmark testing.
+
+    """
+    f = open((BASEPATH / path).resolve())
```

Comment:
I'm not that familiar with `resolve()`. From what I know its better to use the `with open() as f:` syntax. I assume that is still true if using `resolve()`. Do you know?

Reply:
In the code, the resolve() was used to concatenate "Path(__file__).parent" and a string to create a new string represents the path. Now I change the BASEPATH to a string so that we do not need to use resolve() anymore.